e2e-tests/IIoTPlatform-E2E-Tests/Orchestrated/T_TwinBrowseOrchestratedTestTheory.cs

```diff
@@ -0,0 +1,249 @@
+
+                if (!string.IsNullOrEmpty(nodeId)) {
+                    request.AddQueryParameter("nodeId", nodeId);
+                }
+            }
+            else {
+                request.Resource = $"twin/v2/browse/{endpointId}/next";
+                request.AddQueryParameter("continuationToken", continuationToken);
+            }
+
+            var response = await client.ExecuteAsync(request, cts.Token).ConfigureAwait(false);
+            Assert.NotNull(response);
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+
+            dynamic json = JsonConvert.DeserializeObject(response.Content);
+
+            var result = new List<(string NodeId, string NodeClass, bool Children)>();
+
+            foreach (var node in json.references) {
+                result.Add(
+                    (
+                        node.target?.nodeId?.ToString(),
+                        node.target?.nodeClass?.ToString(),
+                        string.Equals(node.target?.children?.ToString(), "true", StringComparison.OrdinalIgnoreCase)));
+            }
+
+            var responseContinuationToken = json.continuationToken;
+
+            if (responseContinuationToken != null) {
+                result.AddRange(await GetNodesAsync(endpointId, null, responseContinuationToken).ConfigureAwait(false));
+            }
+
+            return result;
+        }
+
+        private async Task<List<(string NodeId, string NodeClass, bool Children)>> GetAllNodesAsync(string endpointId, string nodeClass = null, string nodeId = null) {
+            var result = new List<(string NodeId, string NodeClass, bool Children)>();
+            var nodes = await GetNodesAsync(endpointId, nodeId).ConfigureAwait(false);
+
+            foreach (var node in nodes) {
+                if (string.IsNullOrEmpty(nodeClass)
+                    || string.Equals(nodeClass, node.NodeClass, StringComparison.OrdinalIgnoreCase)) {
+                    result.Add(node);
+                }
+                if (node.Children) {
+                    var childNodes = await GetAllNodesAsync(endpointId, nodeClass, node.NodeId).ConfigureAwait(false);
+                    if (childNodes.Any())
+                        _context.OutputHelper.WriteLine($"{node.NodeId}\t{node.NodeClass}\t{childNodes.Count}");
+                    result.AddRange(childNodes);
+                }
+            }
+
+            return result;
+        }
+
+        private struct NodeInfo {
+            public NodeInfo(string nodeClass, string nodeId, bool children = false) : this() {
+                NodeClass = nodeClass;
+                NodeId = nodeId;
+                Children = children;
+            }
+
+            public string NodeClass { get; }
+            public string NodeId { get; }
+            public bool Children { get; }
+        }
+
+        private async Task<List<(string Id, string Url, string ActivationState, string EndpointState)>> ListEndpointsAsync() {
```

Comment:
You could also move this methods to TestHelper if they are also usable/useful by other E2E tests

Reply:
Moved to TestHelper